File: Magna/src/main/java/de/morigm/magna/api/censor/CensorType.java
package de.morigm.magna.api.censor;

public enum CensorType {

    BAN, KICK, WARN, MUTE, DE_OP, SPAWN, GROUP_KICK, GROUP_BAN, KILL, CLEAR, BLIND, NORMAL;

    public static boolean isType(String s) {
        for (CensorType t : values())
            if (t.name().equals(s.toUpperCase()))
                return true;
        return false;
    }

    public static CensorType getType(String s) {
        for (CensorType t : values())
            if (t.name().equals(s.toUpperCase()))
                return t;
        return CensorType.NORMAL;
    }

}

File: Magna/src/main/java/de/morigm/magna/api/censor/CensorType.kt
package de.morigm.magna.api.censor

import java.util.*

enum class CensorType {
    BAN,
    KICK,
    WARN,
    MUTE,
    DE_OP,
    SPAWN,
    GROUP_KICK,
    GROUP_BAN,
    KILL,
    CLEAR,
    BLIND,
    NORMAL;

    companion object {
        fun isType(s: String): Boolean {
            for (t in entries) if (t.name == s.uppercase(Locale.getDefault())) return true
            return false
        }

        @JvmStatic
        fun getType(s: String): CensorType {
            for (t in entries) if (t.name == s.uppercase(Locale.getDefault())) return t
            return NORMAL
        }
    }
}
